### PR TITLE
agent group flag for introspeciton commands

### PR DESCRIPTION
**Pull request by sahil-lakhwani**
### Description of change

- Add support `--agent-group` for introspection commands wherever needed

##### Checklist

- [x] Tested in playground or other setup
- [ ] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [ ] Documentation is changed or added
- [ ] Tests and/or benchmarks are included
- [ ] Breaking changes

### Summary by CodeRabbit

- New Feature: Added the ability to filter and list agents based on specific agent groups. This is achieved through the new `--agent-group` flag added to the `aperturectl agents` and `aperturectl cloud agents` commands.
- New Feature: Enhanced the `aperturectl discovery entities` command with the ability to specify the agent group and the entity to find by name or IP.
- Documentation: Updated the help messages and documentation for the `aperturectl` commands to reflect the new features and usage.
- Refactor: Updated the `ListAgents` and `ListDiscoveryEntities` functions to accept specific request parameters, improving the flexibility and specificity of agent and entity listing.
  
These changes provide users with more control and specificity when listing agents and discovery entities, improving usability and efficiency.

### CODE REVIEW

File: cmd/aperturectl/cmd/utils/agents.go [hunk 1]
```diff
@@ -4,11 +4,13 @@ import (
 	"context"
 	"fmt"
 
-	"google.golang.org/protobuf/types/known/emptypb"
+	cmdv1 "github.com/fluxninja/aperture/v2/api/gen/proto/go/aperture/cmd/v1"
 )
 
-func ListAgents(client IntrospectionClient) error {
-	agents, err := client.ListAgents(context.Background(), &emptypb.Empty{})
+func ListAgents(client IntrospectionClient, agentGroup string) error {
```
Comment (hdkshingala) [inline cmd/aperturectl/cmd/utils/agents.go:10]: Could you please change the cloud agents to also use this method?

File: pkg/agent-functions/agents/agents.go [hunk 2]
```diff
@@ -44,7 +46,7 @@ func (a Agents) ListFlowControlPoints() ([]transport.Result[*cmdv1.ListFlowContr
 	if err != nil {
 		return nil, err
 	}
-	return transport.SendRequests[cmdv1.ListFlowControlPointsAgentResponse](a.etcdTransport, agents, &req)
+	return transport.SendRequests[cmdv1.ListFlowControlPointsAgentResponse](a.etcdTransport, agents["default"], &req)
```
Comment (hdkshingala) [inline pkg/agent-functions/agents/agents.go:49]: Why are we hardcoding default here?

File: cmd/aperturectl/cmd/utils/agents.go (hunk 1 above)
Comment (sahil-lakhwani) [reply cmd/aperturectl/cmd/utils/agents.go:10]: I've changed it [here](https://github.com/fluxninja/aperture/blob/1a15d9e6bcb21428a6474a42deeccdac8e46d1eb/cmd/aperturectl/cmd/cloud/agents.go#L33) if that's what you meant.

Comment (hdkshingala) [reply cmd/aperturectl/cmd/utils/agents.go:10]: No I meant use this method from utils/agents.go instead of having same code at 2 places

Comment (sahil-lakhwani) [reply cmd/aperturectl/cmd/utils/agents.go:10]: Modified.

**Review by hdkshingala — APPROVED** [review]

**Pull request merged**